src/regions/Result.js

```diff
@@ -170,11 +170,11 @@ const Result = types
           });
 
           if (!result) return false;
-          if (choiceValues && !choiceValues.some(v => result.mainValue.includes(v))) return false;
+          if (choiceValues && !choiceValues.some(v => result.mainValue.flat().includes(v))) return false;
```

Comment:
`flat()` is a bit dangerous here, because it may consider selected upper-level same name Taxonomy item enough to submit this related result while the exact needed item is not selected. Could be something like `mainValue.map(v => Array.isArray(v) ? v.at(-1) : v).includes(v)`

Reply:
Good call 👍